Enterprise-CMCS/managed-care-review
# MCR-4217: button and link event tagging for tealium

## Summary
[MCR-4217](https://jiraent.cms.gov/browse/MCR-4217)

- Refactored Tealium implementation
   - Using a provider to configure and intialize Tealium, then provide functions in context for logging.
   - Provider takes in a Tealium client that has all the tealium related functions as a prop.
      - `tealiumClient` is the client used in `prod` and `val` for actual logging to Tealium
      - `devTealiumClient` is the client used in all lower environments where we do not want to log to Tealium.
      - `testTealiumClient` used in `renderWithProviders` for tests.
- `useTealium` no longer initializes Tealium. It provides functions to log user events.
- Added Tealium provider to `renderWithProviders`
- Added two wrapper components for logging in `components/TealiumLogging`
   - `ButtonWithLogging` wrapper around `Button` that takes in logging data as props and generates required logging data from required button props.
   -  `NavLinkWithLogging, LinkWithLogging, ReactRouterLinkWithLogging` Same as the button, but with links. I couldn't figure out the type definition that would allow all three to be combined into one. Did not want to spend any more time trying to figure that out in this PR.
- Add docs on Tealium implementation and how to use it.

#### Related issues

#### Screenshots

#### Test cases covered



## QA guidance

- Please check all links and buttons to make sure they still work and the styling has not changed.
- Please make sure I am logging the right events and that the event data being sent looks correct.
